source/fundamentals/builders.txt

```diff
@@ -0,0 +1,288 @@
+your IDE can provide refactoring support.
+
+Array Operators
+~~~~~~~~~~~~~~~
+
+If your document has properties or fields that serialize to arrays,
+you can use the methods beginning with ``Any``, such as ``AnyEq()`` or
+``AnyLt()``, to compare the entire array against a single item.
+
+Use builders to check which documents in the collection have a
+``Season`` array that includes `"winter":
+
+.. code-block:: csharp
+   :copyable: true
+
+   var builder = Builders<Flower>.Filter;
+   var filter = builder.AnyEq(f => f.Season, "winter");
+
+.. TODO for a complete list of expressions, see the Query page?
+
+Create a Projection
+-------------------
+
+The ``ProjectionDefinitionBuilder`` class provides a type-safe interface for
+defining a projection. Suppose you want to create a projection on the
+``Name`` and ``Price`` fields, but exclude the ``Id`` field.
+
+Use builders to create the projection definition using string-based field names:
+
+.. code-block:: csharp
+   :copyable: true
+
+   var builder = Builders<Flower>.Projection;
+   var projection = builder.Include("Name").Include("Price").Exclude("Id");
+
+You can also use the typed variant to define the projection:
+
+.. code-block:: csharp
+   :copyable: true
+
+   var builder = Builders<Flower>.Projection;
+   var projection = builder.Include(f => f.Name).Include(f => f.Price).Exclude(f => f.Id);
+   
+Finally, you can use the ``Expression()`` method to define the
+projection:
+
+.. code-block:: csharp
+   :copyable: true
+
+   var builder = Builders<Flower>.Projection;
+   var projection = builder.Expression(f => new { Name = f.Name, Price = f.Price });
+
+This definition has a return type of ``ProjectionDefinition<TDocument,
+TProjection>`` whereas the others return a
+``ProjectionDefinition<TDocument>``.
+
+Lambda Expressions
+~~~~~~~~~~~~~~~~~~
+
+The driver supports using expression trees to render projections. A
+lambda expression contains all the information necessary to form both
+the projection on the server as well as the client-side result and
+requires no further information.
+
+When you define a ``Find()`` projection with the ``Expression()``
+method to create a lambda expression, the
+projection is run client-side. The driver inspects the lambda expression
+to determine which fields are referenced and automatically constructs a
+server-side projection to return only those fields.
+
+You can use lambda expressions to create new fields by performing
+operations on values in your documents. The following example shows how
+you can use a lambda expression to project a new ``Profit`` field
+using the ``Price`` and ``Stock`` fields:
+
+.. code-block:: csharp
+   :copyable: true
+
+   var builder = Builders<Flower>.Projection;
+   var projection = builder.Expression(f => new { Profit = f.Price * f.Stock });
+
+The result includes only the ``Profit`` field and automatically excludes
+the``Id`` field. This occurs because the driver inspects the expression
```

Comment:
```suggestion
the ``Id`` field. This occurs because the driver inspects the expression
```

I found this surprising, as the default behavior in other drivers is to never automatically exclude the ``_id`` field. Perhaps this should be an important note.